HelloQuad/HelloQuad.cs

```diff
@@ -0,0 +1,191 @@
+        graphics.SetShaderParameter(ShaderParams.VSP_MODEL, Matrix3x4.IDENTITY);
+        // We set the pixel shader diffuse color to be white. You can change this to 'tint' the texture similar to vertex colors
+        // but this applies to the whole material
+        graphics.SetShaderParameter(ShaderParams.PSP_MATDIFFCOLOR, Color.Blue);
+        // We set cull mode to NONE so our geometry won't be culled (ignored), for this example we don't really need any culling
+        graphics.SetCullMode(CullMode.CULL_NONE);
+
+        // We create a texture from literal data so this code is fully self-contained, you can safely skip the lines below
+        // In your real projects you're most likely going to load texture from the disk using Texture.Load
+        Image image = new Image();
+        image.SetSize(16, 16, 3);
+
+        Color z = Color.Yellow;
+        Color M = Color.Blue;
+        Color k = Color.Black;
+
+        Color[,] imageData =
+        {
+            { k,k,k,k,k,k,k,k,k,k,k,k,k,k,k,k },
+            { k,z,z,z,z,z,z,z,z,z,z,z,z,z,M,k },
+            { k,z,z,z,z,z,z,M,M,z,z,z,z,z,z,k },
+            { k,z,z,z,z,z,z,M,M,z,z,z,z,z,z,k },
+            { k,z,z,z,z,z,M,z,z,M,z,z,z,z,z,k },
+            { k,z,z,z,z,z,M,z,z,M,z,z,z,z,z,k },
+            { k,z,z,z,z,M,z,z,z,z,M,z,z,z,z,k },
+            { k,z,z,z,z,M,z,z,z,z,M,z,z,z,z,k },
+            { k,z,z,z,M,z,z,z,z,z,z,M,z,z,z,k },
+            { k,z,z,z,M,z,z,z,z,z,z,M,z,z,z,k },
+            { k,z,z,M,M,M,M,M,M,M,M,M,M,z,z,k },
+            { k,z,z,M,z,z,z,z,z,z,z,z,M,z,z,k },
+            { k,z,M,z,z,z,z,z,z,z,z,z,z,M,z,k },
+            { k,z,M,z,z,z,z,z,z,z,z,z,z,M,z,k },
+            { k,z,z,z,z,z,z,z,z,z,z,z,z,z,z,k },
+            { k,k,k,k,k,k,k,k,k,k,k,k,k,k,k,k },
+        };
+
+        for (int x = 0; x < 16; x++)
+        {
+            for (int y = 0; y < 16; y++)
+            {
+                image.SetPixel(x, 15 - y, imageData[y, x]);
+            }
+        }
+
+        texture = new Texture2D();
+        texture.SetData(image);
+
+        // We create a new VertexBuffer object, it holds our vertices and is passed to the GPU
+        vertexBuffer = new VertexBuffer();
+        // We set its size and the elements it's containing, the 3rd optional argument (dynamic) should be 'true' if you're planning
+        // to update the VertexBuffer constantly, that will improve performance in those cases.
+        vertexBuffer.SetSize(6, Constants.MASK_POSITION | Constants.MASK_TEXCOORD1, false);
+
+        // Here we lock the vertexBuffer what returns a pointer (IntPtr) to its data (vertexData here), I'm using a code block for clarity
+        System.IntPtr vertexData = vertexBuffer.Lock(0, 6, true);
+        {
+            // We can cast the data pointer to whatever data type we want, here we are only using floats but ideally you will want
+            // to cast it to an object (struct) with properly offsetted fields and maybe unions for things like colors
```

Comment:
I'd like to work out the best method here, likely means using arbitrary vertex declarations, and a bit of work to expose them: https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Source/Atomic/Graphics/GraphicsDefs.h#L150